src/CacheManager.SystemRuntimeCaching/MemoryCacheHandle`1.cs

```diff
@@ -70,6 +73,24 @@ public MemoryCacheHandle(ICacheManagerConfiguration managerConfiguration, CacheH
         public override int Count => (int)_cache.GetCount();
 
         /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string requestedRegion)
+        {
+            return _cache.Select(c => 
+            {
+                bool isToken;
+                bool hasRegion;
+                string region;
+                string key;
+
+                ParseKeyParts(_instanceKeyLength, c.Key, out isToken, out hasRegion, out region, out key);
```

Comment:
After compiling it, any language features disappear, consumers don't have to use VS2017.